test/sun_times_test.exs

```diff
@@ -0,0 +1,60 @@
+ExUnit.start()
+
+defmodule SunTimesTest do
+  use ExUnit.Case
+
+  defmodule TestHelpers do
+    def testDate() do
+      DateTime.new!(~D[2022-05-26], ~T[00:00:00], "Etc/UTC")
+    end
+
+    def compareUTCArrays(arr1, arr2, tolerance) do
+      -11..11
+      |> Enum.map(fn hour ->
+        assert abs(
+                 DateTime.diff(
+                   Enum.at(arr1, hour),
+                   Enum.at(arr2, hour),
+                   :second
+                 )
+               ) <= tolerance
+      end)
+    end
+
+    def loopThroughTimeZones(lat, lon, event, offset) do
+      if event == :set do
+        -11..11
+        |> Enum.map(fn hour ->
+          SunTimes.set(testDate(), lat, lon + (hour + 1) * 15)
+          |> DateTime.add(-offset * 3600, :second)
+        end)
+      else
+        -11..11
+        |> Enum.map(fn hour ->
+          SunTimes.rise(testDate(), lat, lon + (hour + 1) * 15)
+          |> DateTime.add(-offset * 3600, :second)
+        end)
+      end
+    end
+  end
+
+  describe "rise/3" do
+    test "offsets of +/- 15 degrees should adjust sunrise by aprox 1 hour" do
+      assert TestHelpers.compareUTCArrays(
+               TestHelpers.loopThroughTimeZones(30, 5, :rise, 0),
+               TestHelpers.loopThroughTimeZones(30, 20, :rise, -1),
```

Comment:
This is off. You need to check the mental invariant: "Am I asking for a longitude outside the range [-180, 180]? If you're going from -11 to 11, then you want longitudes in the rang [-15, 15]. You can fix this by comparing -10 to 5, for example.